heft-plugins/heft-jest-plugin/src/schemas/heft-jest-plugin.schema.json

```diff
@@ -11,6 +11,11 @@
       "title": "Disable Configuration Module Resolution",
       "description": "If set to true, modules specified in the Jest configuration will be resolved using Jest default (rootDir-relative) resolution. Otherwise, modules will be resolved using Node module resolution.",
       "type": "boolean"
+    },
+    "enableNodeEnvManagement": {
+      "title": "Enable management of the NODE_ENV variable",
+      "description": "If set to false, heft-jest-plugin will not set or unset the NODE_ENV variable. Otherwise, NODE_ENV will be set to `test` before execution and cleared after. If the NODE_ENV value is already set to a value that is not `test`, warning message appears.",
+      "type": "boolean"
```

Comment:
We generally try to design `boolean` options that default to `false` rather than `true`.

For example we could invert the logic and call the setting something like `preserveNodeEnvVariable` (defaults to false, `true` means don't change `NODE_ENV`.)

(Although TBH the behavior you're implementing in this PR seems very sensible, so I don't care much whether we have a setting to disable it. @iclanton did you suggest that?)

Reply:
Otherwise this PR looks good 👍

Reply:
@woss what do you think?

Reply:
@octogonz I think this is a good enough approach. it is clear and concise and follows the jest approach when it comes to the NODE_ENV variable. 

I suggest merging this and then we can see if there is anyone having a hard time understanding or setting this up. If that happens I volunteer to improve this functionality according to solutions we agree or you provide 🙂